utils/test_helpers.go

```diff
@@ -1,27 +1,228 @@
 package utils
 
 import (
+	"bytes"
+	"errors"
 	"fmt"
-	fuzz "github.com/AdaLogics/go-fuzz-headers"
+	"os"
 	"reflect"
-	"testing"
+	"runtime"
+	"runtime/debug"
+	"sync"
+	"time"
+
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
 )
 
+type T struct {
+	parent   *T
+	name     string
+	mu       sync.RWMutex
+	skipped  bool
+	failed   bool
+	finished bool
+	cleanups []func()
+	output   bytes.Buffer
+}
+
+// Most of the T functions are copied from the stdlib
+
+var errNilPanicOrGoexit = errors.New("test executed panic(nil) or runtime.Goexit")
+
+type panicHandling int
+
+const (
+	normalPanic panicHandling = iota
+	recoverAndReturnPanic
+)
+
+func (t *T) log(s string) { _, _ = t.output.WriteString(s) }
```

Comment:
For libFuzzer it is better to print out the string immediately instead of storing it in a string builder and print it out later. So could we do `fmt.Printlf(s)` instead of `t.output.WriteString(s)`?

Reply:
@AdamKorcz `Printf` seems wrong because `s` may contain things like `%s`. In any case, it seems we can just remove this helper function.